python-avd/pyavd/_eos_designs/structured_config/core_interfaces_and_l3_edge/ethernet_interfaces.py

```diff
@@ -20,50 +20,26 @@ class EthernetInterfacesMixin(Protocol):
-        ethernet_interfaces = []
-
+    @structured_config_contributor
+    def ethernet_interfaces(self: AvdStructuredConfigCoreInterfacesAndL3EdgeProtocol) -> None:
+        """Set the structured config for ethernet_interfaces."""
         for p2p_link, p2p_link_data in self._filtered_p2p_links:
+            ethernet_interface = EosCliConfigGen.EthernetInterfacesItem()
             if p2p_link_data["port_channel_id"] is None:
                 # Ethernet interface
-                ethernet_interface = self._get_common_interface_cfg(p2p_link, p2p_link_data)
-                ethernet_interface["description"] = self._p2p_link_ethernet_description(p2p_link_data)
-                ethernet_interface.update(self._get_ethernet_cfg(p2p_link))
-
-                # Remove None values
-                ethernet_interface = {key: value for key, value in ethernet_interface.items() if value is not None}
-
-                append_if_not_duplicate(
-                    list_of_dicts=ethernet_interfaces,
-                    primary_key="name",
-                    new_dict=ethernet_interface,
-                    context=f"Ethernet Interfaces defined under {self.data_model} p2p_link",
-                    context_keys=["name", "peer", "peer_interface"],
-                )
+                self._set_common_interface_cfg(p2p_link, p2p_link_data, ethernet_interface)
+                ethernet_interface.ptp = self._get_ptp_config_interface(p2p_link, output_type=EosCliConfigGen.EthernetInterfacesItem.Ptp)
+                ethernet_interface.description = self._p2p_link_ethernet_description(p2p_link_data)
+                ethernet_interface.speed = p2p_link.speed
+                self.structured_config.ethernet_interfaces.append(ethernet_interface)
 
             # Port-Channel members
             for member in p2p_link_data["port_channel_members"]:
-                ethernet_interface = self._get_port_channel_member_cfg(p2p_link, p2p_link_data, member)
-                ethernet_interface["description"] = self._port_channel_member_description(p2p_link_data, member)
-                ethernet_interface.update(self._get_ethernet_cfg(p2p_link))
-
-                # Remove None values
-                ethernet_interface = {key: value for key, value in ethernet_interface.items() if value is not None}
-
-                append_if_not_duplicate(
-                    list_of_dicts=ethernet_interfaces,
-                    primary_key="name",
-                    new_dict=ethernet_interface,
-                    context=f"Ethernet Interfaces defined under {self.data_model} p2p_link port-Channel members",
-                    context_keys=["name", "peer", "peer_interface"],
-                )
-
-        if ethernet_interfaces:
-            return ethernet_interfaces
-
-        return None
+                ethernet_interface = EosCliConfigGen.EthernetInterfacesItem()
+                self._set_port_channel_member_cfg(p2p_link, p2p_link_data, member, ethernet_interface)
```

Comment:
this should be called `_update` as you are updating the object rather than setting the structured_config
```suggestion
                self._set_port_channel_member_cfg(p2p_link, p2p_link_data, member, ethernet_interface)
```

Reply:
actually - just set directly the interface in structured_configuration in _set_port_channel_memebr_config (which I think you can bring back in this `ethernet_interfaces.py` file for more clarity as it is not used anywhere else

Reply:
Moved these parameters inside the port-channel-members for loop in ethernet_interfaces.py (removed this method)